### DISCUSSION TITLE
Can i build this package in windows?

### DISCUSSION THREAD

**Discussion by Hamidrzash** [scalar/scalar#4257] [Q&A]
Can i build this package in windows?

Failed:    @scalar/build-tooling#build

**Comment by hanspagel** [comment]
What’s the error?

**Comment by Hamidrzash** [comment]
after pnpm install

`pnpm turbo build
╭──────────────────────────────────────────────────────────────────────────╮
│                                                                          │
│                     Update available v2.1.2 ≫ v2.3.3                     │
│    Changelog: https://github.com/vercel/turborepo/releases/tag/v2.3.3    │
│             Run "npx @turbo/codemod@latest update" to update             │
│                                                                          │
│          Follow @turborepo for updates: https://x.com/turborepo          │
╰──────────────────────────────────────────────────────────────────────────╯
turbo 2.1.2

• Packages in scope: @scalar-examples/cdn-api-reference, @scalar-examples/docusaurus, @scalar-examples/express-api-reference, @scalar-examples/fastify-api-reference, @scalar-examples/nestjs-api-reference-express, @scalar-examples/nestjs-api-reference-fastify, @scalar-examples/nextjs-api-reference, @scalar-examples/proxy-server, @scalar-examples/react, @scalar-examples/snippetz, @scalar-examples/ssg, @scalar-examples/web, @scalar/api-client, @scalar/api-client-proxy, @scalar/api-client-react, @scalar/api-reference, @scalar/api-reference-editor, @scalar/api-reference-react, @scalar/aspnetcore, @scalar/build-tooling, @scalar/cli, @scalar/code-highlight, @scalar/components, @scalar/config, @scalar/docusaurus, @scalar/draggable, @scalar/echo-server, @scalar/express-api-reference, @scalar/fastify-api-reference, @scalar/fetch, @scalar/galaxy, @scalar/hono-api-reference, @scalar/icons, @scalar/import, @scalar/mock-server, @scalar/nestjs-api-reference, @scalar/nextjs-api-reference, @scalar/nextjs-openapi, @scalar/nuxt, @scalar/oas-utils, @scalar/object-utils, @scalar/openapi-parser, @scalar/openapi-types, @scalar/play-button, @scalar/postman-to-openapi, @scalar/snippetz, @scalar/themes, @scalar/ts-to-openapi, @scalar/types, @scalar/use-codemirror, @scalar/use-hooks, @scalar/use-toasts, @scalar/use-tooltip, @scalar/void-server, @scalarapi/api-reference, client-scalar-com, playwright, scalar-app
• Running build in 58 packages
• Remote caching disabled
@scalar/echo-server:build: cache miss, executing e41af6ac8cedd87b
@scalar/build-tooling:build: cache miss, executing 123d572917c96301
@scalar/draggable:build: cache miss, executing 2918dae760089c75
@scalarapi/api-reference:build: cache hit, replaying logs c40c04a03d4daf06
@scalarapi/api-reference:build:
@scalarapi/api-reference:build: > @scalarapi/api-reference@0.1.1 build D:\scalar\packages\api-reference\docker
@scalarapi/api-reference:build: > command -v docker >/dev/null 2>&1 && docker build -t scalarapi/api-reference:latest ./src || echo '⚠️ Docker is not available, skipping the build.'
@scalarapi/api-reference:build:
@scalarapi/api-reference:build: The system cannot find the path specified.
@scalarapi/api-reference:build: '?? Docker is not available, skipping the build.'
@scalar/echo-server:build:
@scalar/echo-server:build: > @scalar/echo-server@0.7.2 build D:\scalar\packages\echo-server
@scalar/echo-server:build: > vite build && pnpm types:build && tsc-alias -p tsconfig.build.json
@scalar/echo-server:build:
@scalar/build-tooling:build:
@scalar/build-tooling:build: > @scalar/build-tooling@0.1.12 build D:\scalar\packages\build-tooling
@scalar/build-tooling:build: > bash scripts/build-rollup.sh
@scalar/build-tooling:build:
@scalar/draggable:build:
@scalar/draggable:build: > @scalar/draggable@0.1.7 build D:\scalar\packages\draggable
@scalar/draggable:build: > vite build && pnpm types:build && tsc-alias -p tsconfig.build.json
@scalar/draggable:build:
@scalar/echo-server:build: vite v5.4.10 building for production...
                                                                  @scalar/echo-server:build: transforming...
                                                                                                            @scalar/echo-server:build: ✓ 2 modules transformed.
                                       @scalar/echo-server:build: rendering chunks...
                                                                                     @scalar/echo-server:build: computing gzip size...
              @scalar/echo-server:build: dist/index.js  1.69 kB │ gzip: 0.64 kB
                                                                               @scalar/echo-server:build: dist/index.cjs  1.06 kB │ gzip: 0.58 kB
                         @scalar/echo-server:build: ✓ built in 158ms
                                                                    @scalar/echo-server:build:
                                                                                               @scalar/echo-server:build: > @scalar/echo-server@0.7.2 types:build D:\scalar\packages\echo-server
                                                                        @scalar/echo-server:build: > tsc -p tsconfig.build.json
       @scalar/echo-server:build:
                                  @scalar/build-tooling:build: <3>WSL (11) ERROR: CreateProcessEntryCommon:505: execvpe /bin/bash failed 2
                  @scalar/build-tooling:build: <3>WSL (11) ERROR: CreateProcessEntryCommon:508: Create process not expected to return
             @scalar/draggable:build: vite v5.4.10 building for production...
                                                                             @scalar/build-tooling:build:  ELIFECYCLE  Command failed with exit code 1.
@scalar/build-tooling:build: ERROR: command finished with error: command (D:\scalar\packages\build-tooling) D:\Program Files\nodejs\pnpm.CMD run build exited (1)
@scalar/build-tooling#build: command (D:\scalar\packages\build-tooling) D:\Program Files\nodejs\pnpm.CMD run build exited (1)

 Tasks:    1 successful, 4 total
Cached:    1 cached, 4 total
  Time:    9.492s
Failed:    @scalar/build-tooling#build

 ERROR  run failed: command  exited (1)`

**Comment by hanspagel** [comment]
Ah, okay. Bash scripts don’t work on Windows by default. Maybe enabling WSL helps?

What’s your motivation to build the package locally?

**Comment by Hamidrzash** [comment]
WSL is enabled.
I want to contribute.

**Comment by Diggsey** [comment]
I was looking to work on https://github.com/scalar/scalar/issues/4975 but ran into the same issue. Unfortunately it's probably too big a task to replace all the shell scripts with a more portable solution...

**Comment by xC0dex** [comment]
I created an issue for this #5328!

**Comment by hanspagel** [comment]
this should work for a while already
